### PR TITLE
boot: misc UC20 changes

### PR DESCRIPTION
**Pull request by anonymouse64**
* Use the boot status variables in more places, especially tests and in snap-bootstrap
* Add doc-comments to the ExtractedRunKernelImageBootloader interface
* Change `TryKernel()` to return 2 values, the snap and an error, and the error can be `bootloader.ErrNoTryKernelRef` if the try-kernel doesn't exist; refactor to reflect this
* Do some logging to stderr in `snap-bootstrap initramfs-mounts` to resolve some TODO:UC20's

### CODE REVIEW

File: bootloader/bootloader.go [hunk 1]
```diff
@@ -33,6 +33,10 @@ import (
 var (
 	// ErrBootloader is returned if the bootloader can not be determined.
 	ErrBootloader = errors.New("cannot determine bootloader")
+
+	// ErrNoTryKernelRef is returned if the bootloader cannot find an enabled
+	// try-kernel.
```
Comment (pedronis) [inline bootloader/bootloader.go]: finds no enabled try-kernel.

File: bootloader/bootloader.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ import (
...
+	ErrNoTryKernelRef = errors.New("cannot find try-kernel")
```
Comment (pedronis) [inline bootloader/bootloader.go]: no try-kernel referenced

File: bootloader/bootloader.go [hunk 3]
```diff
@@ -95,13 +95,42 @@ type RecoveryAwareBootloader interface {
...
 	Bootloader
-	EnableKernel(snap.PlaceInfo) error    // makes the symlink
-	EnableTryKernel(snap.PlaceInfo) error // makes the symlink
-	Kernel() (snap.PlaceInfo, error)      // gives the symlink
-	TryKernel() (snap.PlaceInfo, error)   // gives the symlink (if exists)
-	DisableTryKernel() error              // removes the symlink
+
+	// EnableKernel enables the specified kernel on ubuntu-boot to be used
+	// during normal boots. The specified snap should already have been
+	// extracted.
```
Comment (pedronis) [inline bootloader/bootloader.go]: this should mention symlinking as possible implementation like EnableTryKernel does.

Comment (bboozzoo) [reply bootloader/bootloader.go]: Fixed now. Rephrased this and EnableTryKernel, felt like wording was a  little off.

**Pull request merged**